programs/uxd/tests/unit_tests/utils/calculate_depositories_mint_collateral_amount.rs

```diff
@@ -29,27 +33,39 @@ mod test_calculate_depositories_mint_collateral_amount {
             ui_to_native_amount(700_000);
 
         // Compute
+        let depositories_target_and_redeemable_under_management = vec![
```

Comment:
You could even create a test structure to loop over it and avoid manual configuration, example:

```
    struct TestCase {
        requested_amount: u64,
        depository_info_list:
            Vec<DepositoryTargetRedeemableAmountAndRedeemableAmountUnderManagement>,
        expected_mint_amounts: Vec<u64>,
    }

    #[test]
    fn test_with_simplest_case() -> Result<()> {
        let test_case = TestCase {
            requested_amount: ui_to_native_amount(1_000_000),
            depository_info_list: vec![
                DepositoryTargetRedeemableAmountAndRedeemableAmountUnderManagement {
                    target_redeemable_amount: ui_to_native_amount(1_000_000),
                    redeemable_amount_under_management: ui_to_native_amount(500_000).into(),
                },
                DepositoryTargetRedeemableAmountAndRedeemableAmountUnderManagement {
                    target_redeemable_amount: ui_to_native_amount(1_000_000),
                    redeemable_amount_under_management: ui_to_native_amount(800_000).into(),
                },
                DepositoryTargetRedeemableAmountAndRedeemableAmountUnderManagement {
                    target_redeemable_amount: ui_to_native_amount(1_000_000),
                    redeemable_amount_under_management: ui_to_native_amount(700_000).into(),
                },
            ],
            expected_mint_amounts: vec![
                ui_to_native_amount(500_000),
                ui_to_native_amount(200_000),
                ui_to_native_amount(300_000),
            ],
        };

        // Compute
        let amounts = calculate_depositories_mint_collateral_amount(
            test_case.requested_amount,
            &test_case.depository_info_list,
        )?;

        for (i, expected_amount) in test_case.expected_mint_amounts.iter().enumerate() {
            assert_eq!(amounts[i], *expected_amount);
        }

        Ok(())
    }
```

Reply:
Good call!
Ima stick with the manual method for now as it adds a bit of complexity to use a TestCase struct and loop on it.
We can re-introduce this if we start having hundreds of tests, I like the line-by-line asserts instead of an assert loop.

Definitly going to apply the first comment tho, thanks!